source/documentBase.py

```diff
@@ -86,6 +86,15 @@ def _getTableCellAt(self,tableID,startPos,row,column):
 		"""
 		raise NotImplementedError
 
+	def _getTableDimensions(self,tableID):
+		"""
+		Returns height and width of the table.
+		@param tableID: the ID of the table.
+		@returns: Height and width of the table.
+		@rtype: L{Tuple[int, int]}

```

Comment:
For new code types for parameters and return values should be specified using Python type annotations if possible. The same goes for all docstrings below - I'm not going to comment on all of them.

Reply:
Fixed that. The only thing I couldn't annotate is table_id as it is a union. This union is properly defined in #13345 - will annotate once it lands.